x-pack/plugins/security_solution/public/management/pages/policy/view/event_filters/flyout/policy_event_filters_flyout.test.tsx

```diff
@@ -0,0 +1,296 @@
+import {
+  AppContextTestRender,
+  createAppRootMockRenderer,
+} from '../../../../../../common/mock/endpoint';
+import { EndpointDocGenerator } from '../../../../../../../common/endpoint/generate_data';
+import { PolicyData } from '../../../../../../../common/endpoint/types';
+import { getPolicyEventFiltersPath } from '../../../../../common/routing';
+import { eventFiltersListQueryHttpMock } from '../../../../event_filters/test_utils';
+import { PolicyEventFiltersFlyout } from './policy_event_filters_flyout';
+import { parseQueryFilterToKQL, parsePoliciesAndFilterToKql } from '../../../../../common/utils';
+import { SEARCHABLE_FIELDS } from '../../../../event_filters/constants';
+import {
+  FoundExceptionListItemSchema,
+  UpdateExceptionListItemSchema,
+} from '@kbn/securitysolution-io-ts-list-types';
+
+const endpointGenerator = new EndpointDocGenerator('seed');
+const getDefaultQueryParameters = (customFilter: string | undefined = '') => ({
+  path: '/api/exception_lists/items/_find',
+  query: {
+    filter: customFilter,
+    list_id: ['endpoint_event_filters'],
+    namespace_type: ['agnostic'],
+    page: undefined,
+    per_page: 100,
+    sort_field: undefined,
+    sort_order: undefined,
+  },
+});
+const emptyList = {
+  data: [],
+  page: 1,
+  per_page: 10,
+  total: 0,
+};
+
+const getCleanedExceptionWithNewTags = (
+  exception: UpdateExceptionListItemSchema,
+  testTags: string[],
+  policy: PolicyData
+) => {
+  const exceptionToUpdateCleaned = {
+    ...exception,
+    tags: [...testTags, `policy:${policy.id}`],
+  };
+  // Clean unnecessary fields for update action
+  [
+    'created_at',
+    'created_by',
+    'created_at',
+    'created_by',
+    'list_id',
+    'tie_breaker_id',
+    'updated_at',
+    'updated_by',
+  ].forEach((field) => {
+    delete exceptionToUpdateCleaned[field as keyof UpdateExceptionListItemSchema];
+  });
+  return exceptionToUpdateCleaned;
+};
+
+describe('Policy details event filters flyout', () => {
+  let render: () => Promise<ReturnType<AppContextTestRender['render']>>;
+  let renderResult: ReturnType<AppContextTestRender['render']>;
+  let history: AppContextTestRender['history'];
+  let mockedContext: AppContextTestRender;
+  let mockedApi: ReturnType<typeof eventFiltersListQueryHttpMock>;
+  let policy: PolicyData;
+  let onCloseMock: jest.Mock;
+
+  beforeEach(() => {
+    policy = endpointGenerator.generatePolicyPackagePolicy();
+    mockedContext = createAppRootMockRenderer();
+    mockedApi = eventFiltersListQueryHttpMock(mockedContext.coreStart.http);
+    onCloseMock = jest.fn();
+    ({ history } = mockedContext);
+    render = async () => {
+      await act(async () => {
+        renderResult = mockedContext.render(
+          <PolicyEventFiltersFlyout policyItem={policy} onClose={onCloseMock} />
+        );
+        await waitFor(mockedApi.responseProvider.eventFiltersList);
```

Comment:
I'm not sure this is doing what you expect it to be done.  What is the intent - to wait for the data to be returned? If so, then you need to check for when it is actually called. maybe something like this:

```typescript
await waitFor(
  () => expect(mockedApi.responseProvider.eventFiltersList).toHaveBeenCalled()
);
```

Reply:
yeah. exactly. It was what I was looking for, thanks!